pkg/skaffold/build/buildpacks/lifecycle.go

```diff
@@ -91,7 +91,7 @@ func (b *Builder) build(ctx context.Context, out io.Writer, a *latestV1.Artifact
 
 	builderImage, runImage, pullPolicy := resolveDependencyImages(artifact, b.artifacts, a.Dependencies, b.pushImages)
 
-	if err := runPackBuildFunc(ctx, output.GetWriter(out), b.localDocker, pack.BuildOptions{
+	if err := runPackBuildFunc(ctx, output.GetUnderlyingWriter(out), b.localDocker, pack.BuildOptions{
```

Comment:
So doesn't this mean the output from this command won't go through our event logging?

Maybe we do this if there is no event writer?

Reply:
I was actually curious as to why we did this, and was going to address this in a follow up, as I believe we should be able to use our writer here. I'll probably need to make some more changes to the output setup that's happening in these functions

Reply:
The other thing to look at is the https://github.com/creack/pty package